engine/src/main/java/com/sandboni/core/engine/sta/connector/TestSuiteConnector.java

```diff
@@ -0,0 +1,36 @@
+package com.sandboni.core.engine.sta.connector;
+
+import com.sandboni.core.engine.sta.Context;
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.TestSuiteVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Connects between each suite test class to it's related methods (test vertices)
+ */
+public class TestSuiteConnector implements Connector {
+
+    @Override
+    public void connect(Context context) {
+        Set<Vertex> tests = context.getLinks().filter(l -> l.getLinkType() == LinkType.ENTRY_POINT).map(Link::getCallee).collect(Collectors.toSet());
+        Set<Vertex> allTestSuiteVertices = context.getLinks().filter(l -> l.getLinkType() == LinkType.TEST_SUITE).map(Link::getCallee).collect(Collectors.toSet());
+        // connect test suite to relevant test (if applicable):
+        tests.parallelStream().forEach(t -> {
```

Comment:
Before iterating, maybe would be better practice to check if we have any Test Suite vertices in `allTestSuiteVertices`

Reply:
handled in proceed() method